waspc/src/Wasp/Generator/ServerGenerator.hs

```diff
@@ -179,12 +191,14 @@ npmDepsForWasp spec =
             ("typescript", "^5.1.0"),
             ("@types/express", "^4.17.13"),
             ("@types/express-serve-static-core", "^4.17.13"),
-            ("@types/node", "^18.11.9"),
-            ("@tsconfig/node" ++ show (major NodeVersion.latestMajorNodeVersion), "^1.0.1"),
+            ("@types/node", "^" <> majorNodeVersionStr <> ".0.0"), -- TODO: This is challenging hm, can we do something better?
+            ("@tsconfig/node" <> majorNodeVersionStr, "^1.0.1"),
```

Comment:
@sodic @infomiho I am not sure about this here -> We don't really know which node version they are using, so it feels a bit tricky to specify node version here!

I set these both to major version of the node version that we use as a lower bound, so in this case `20`. I think this is an ok choice.

But, what if they use Node 21? Is it then bad that we have types for Node 20? Probalby not terrible, but also not great. But how can we go about this anyway, I don't see a way to solve this? Maybe restructuring will help with this?

Reply:
Ideally, we get the `userNodeVersion` and use that? 

If his version is okay with us, then we just match the types version and the `tsconfig.json` version. Maybe.

A problem with that is the deployment -> user uses Node 21 in dev, but then we deploy it with Node 20. That could be weird?

Reply:
Hm!
So let's say we use their version of node -> what if one time they use Node 20, and next time they use Node 21?

How would this work in a typical npm project -> I guess user would pin down a specific version of Node if they are using types/node (via package.json's `engines` field)? They probably wouldn't say "this project can work with any node >=20"? Or what would they do?

So I guess we need to know which NodeJS they want to use. I see multiple ways to go about it:
1. We detect which node they are using and use that.
2. We let them specify, in .wasp, which node version they want to use.
3. With restructuring, maybe this situation goes away @sodic ?

I guess, if we know which node version they want to use, we just take a major number of that one and use it in these places.

Another question: what if we just stick to what I did here in this PR? What if we specify node 20 in these places and they use 21 -> I guess that shoulnd't be a big issue? But could it be an issue? In which situations?

Reply:
`nvmrc` just says which specific node version to use, right?

What I am curious about here is how do we know which version of node types to import here  - should it dynamically change according to the node version they are currently using while generating these files, or should they specify node version somehow in advance (ok nvrmc could come into play here I guess), or do we pin down some version regardless of their node version (that sounds wrong, but might work in practice?).

I will take a better look at other projects, but I think this might be a problem unique to use due to the fact that we have "framework" code that is nor part of our own npm package.

I wonder if answering a more general question might be beneficial here: let's say I have  a TS project that I am working on with my teammates. Should I always pin down the node version used for it (via `engines` in package.json), to make sure the whole team uses the same node version, or should I allow e.g. all versions >= 18? Sounds to me like if I am developing an app and not a library, that it would be better to have that version fixed. If I am developing a library, well then I probably want to test for multiple versions of nodejs (all that library supports) and that makes it a bit more complicated, and I guess I would also have hard time choosing which node version to specify for @types/node?
Or if I am developing an app and want to support different versions of node, I think then I would have the same problem. So I wonder how this problem is solved in those situations.

I asked ChatGPT just in case it offers some interesting opinion: the best thing it came up with was that it is probably best bet to choose the lowest version of node we support, since node releases are backwards compatible so while some new features might be missing in the types, at least we don't have "redundant" types / features.

To finish my train of thought here, I think we have two questions here:
1. Should we allow Wasp devs to choose which npm version they want to use for their Wasp project, in some range of supported versions of course (e.g. >= latest LTS), or do we say "for Wasp version X you need to use node version Y" like we do now and that is it?
2. If they can choose the node version they want to use, do we force them to pick a specific version from that range and stick to it so the whole team is using the same version, or can they choose any version from that range as they like when they like? While making this decision, we should probably consider what is the recommended practice when building web apps (not libs, because advice is probably the opposite).